### ISSUE TITLE
dotnet run should no suppress console output

### ISSUE THREAD

**Issue by RehanSaeed** [dotnet/cli#7932]
## Steps to reproduce

I'm using `dotnet watch run` on my Vue.js SPA project I built using the ASP.NET Core SPA templates. I added TS-Lint support to webpack via the `tslint-loader`. My csproj contains the following code (default to the SPA template) that runs webpack before a build:

```
<Target Name="DebugRunWebpack" BeforeTargets="Build" Condition=" '$(Configuration)' == 'Debug' ">
    <!-- Ensure Node.js is installed -->
    <Exec Command="node --version" ContinueOnError="true">
      <Output TaskParameter="ExitCode" PropertyName="ErrorCode" />
    </Exec>
    <Error Condition="'$(ErrorCode)' != '0'" Text="Node.js is required to build and run this project. To continue, please install Node.js from https://nodejs.org/, and then restart your command prompt or IDE." />

    <!-- In development, the dist files won't exist on the first run or when cloning to
         a different machine, so rebuild them if not already present. -->
    <Message Importance="high" Text="Performing first-run Webpack build..." />
    <Exec Command="node node_modules/webpack/bin/webpack.js --config webpack.config.vendor.js" />
    <Exec Command="node node_modules/webpack/bin/webpack.js" />
  </Target>
```

Using `dotnet run` or more importantly `dotnet watch run` in my case, supresses all console output, so I can't see any TS-Lint errors.

## Expected  behavior

`dotnet run` should not suppress console output or at least have an option to no do so.

## Actual behavior

`dotnet run` supresses console output.

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.0.2)

Product Information:
 Version:            2.0.2
 Commit SHA-1 hash:  a04b4bf512

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.2\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d
```

**Comment by livarcocc** [comment]
Does this repro at all with dotnet run or only with dotnet watch run?

**Comment by RehanSaeed** [comment]
I get no TS-Lint output when running under `dotnet run` or `dotnet watch run` but do get it from `dotnet build`.

**Comment by livarcocc** [comment]
I am not familiar with TS-Lint. Is it run only during build and not by your application output explicitly? This behavior is by design.

We do not display the build content during run because we want to display only application output, instead of polluting the console with the build output.

**Comment by RehanSaeed** [comment]
If you look at the MSBuild snippet above, webpack is configured to run before a build. You also don't see Webpack console output, so this is not related to TS-Lint.

My personal use case is to see TS-Lint errors. I'm using a TS-Lint loader for Webpack, so it runs as part of the Webpack process.

**Comment by RehanSaeed** [comment]
Could a command line switch be added to enable build output? This is a pretty common use case.

**Comment by livarcocc** [comment]
My point is that we don't display build output during run. We do so so that the dotnet run output matches the same output as dotnet <app.dll>, that you use when invoking your app.

**Comment by RehanSaeed** [comment]
Additionally, for a single page application, console output from ASP.NET Core is a one time thing (unless you are also hosting an API and doing AJAX calls to it) and then irrelevant from then on as the browser has all the JavaScript files it needs.

**Comment by svick** [comment]
@RehanSaeed Would a workaround be acceptable for you?

For example, if you add `<Target Name="MyRun" DependsOnTargets="Build"><Exec Command="dotnet run" /></Target>` to your csproj, then executing `dotnet watch msbuild /t:MyRun` will build and run your project, and include the build output.

**Comment by livarcocc** [comment]
@RehanSaeed I don't think it makes too much sense for the CLI to custom tailor the output of dotnet run per application type. Strictly speaking, even similar applications, like console, may have different  output needs for dotnet run.

We optimized for what we believed was the most common one: make it match the output of running my application. In fact, we haven't had many people asking for the build output during run, quite the opposite. That was our behavior a while ago and we changed due to community feedback.

I would go ahead and try @svick customization above. Thanks for it @svick.

@KathleenDollard what do you think of adding a verbosity flag to run to match build verbosity?

**Comment by livarcocc** [comment]
@peterhuene you are already adding this in your PR for --no-build meaning no implicit restore.

**Comment by peterhuene** [comment]
@RehanSaeed Thanks for reporting this issue!  With the fix currently in #8100, you should be able to pass the `-v:<verbosity>` option to enable the restore/build output.
